public/apps/configuration/utils/password-edit-panel.tsx

```diff
@@ -0,0 +1,77 @@
+/*
+ *   Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import React, { useState } from 'react';
+import { EuiFieldText, EuiIcon } from '@elastic/eui';
+import { FormRow } from './form-row';
+
+// At least one uppercase, one lowercase, one digit and one special character
+// and at least 8 characters long
+const PASSWORD_PATTERN = /^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?([^\w\s]|[_])).{8,}$/;
+
+export function PasswordEditPanel(props: {
+  updatePassword: (p: string) => void;
+  updateIsInvalid: (v: boolean) => void;
+}) {
+  const [password, setPassword] = useState<string>('');
+  const [isPasswordInvalid, setIsPasswordInvalid] = useState<boolean>(false);
+  const [isRepeatPasswordInvalid, setIsRepeatPasswordInvalid] = useState<boolean>(false);
+
+  const updateValidStatus = () => {
+    props.updateIsInvalid(isPasswordInvalid || isRepeatPasswordInvalid);
+  };
+
+  const passwordChangeHandler = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const newValue = e.target.value;
+    setPassword(newValue);
+    setIsPasswordInvalid(!PASSWORD_PATTERN.test(newValue) && newValue !== '');
+    props.updatePassword(newValue);
+    updateValidStatus();
```

Comment:
Any reason to create a one line function instead of directly calling it same as the line 40?

Reply:
Thought to be a several lines one, will remove.